### PR TITLE
Reduce use of stateful components 

### PR DESCRIPTION
**Pull request by NickLewanowicz**
## Description

I went through all the component folders and changed any component that used React.Component and didnt use state/hooks to just be a stateless component.

## Related Issue

#24 

## Motivation and Context

This will hopfully reduce the complexity of our codebase and make it easier to learn

## How Has This Been Tested?

Ran test suite and did manual testing

### CODE REVIEW

File: src/App/Home/Project/Project.js [hunk 1]
```diff
@@ -13,36 +13,33 @@ const defaultState = {
...
-	shouldComponentUpdate(nextProps, nextState) {
-		if (nextProps.data.project) {
-			nextState.items = this.getItems(nextProps.data.project)
-		}
-		return true
+
+	static getDerivedStateFromProps(props, state) {
+		state.items = props.data.project? getItems(props.data.project) : []
+		return state
 	}
```
Comment (NickLewanowicz) [inline src/App/Home/Project/Project.js]: Using getDerivedStateFromProps makes more sense then shouldComponentUpdate

File: src/App/Home/Project/Project.js [hunk 2]
```diff
@@ -13,36 +13,33 @@ const defaultState = {
...
-	getItems(project) {
-		if (!project) { return [] }
-		return [].concat(...project.sections.map(section => {
-			return [].concat({ type: 'section', data: section }, ...section.tasks.map(task => {
-				return { type: 'task', data: task }
-			}))
-		}))
-	}
-
-
```
Comment (NickLewanowicz) [inline src/App/Home/Project/Project.js]: Moved out of the component as this is a generic functional piece

File: src/App/Home/Project/Project.js [hunk 3]
```diff
@@ -81,8 +78,7 @@ class Project extends Component {
 					<EditableTask
 						item={this.state.selectedItem}
 						onChange={this.onChange}
-						selectItem={this.selectItem}
-						location={this.props.location.pathname} />
```
Comment (Foopert) [inline src/App/Home/Project/Project.js]: Why was location removed?

Comment (NickLewanowicz) [reply src/App/Home/Project/Project.js]: Its not used for anything

**Pull request merged**